packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -19,18 +14,47 @@ export interface SerializedAttributionCollection {
     length: number;
 }
 
-export class AttributionCollection<T> {
-    private readonly entries: RedBlackTree<number, T> = new RedBlackTree(compareNumbers);
+export interface IAttributionCollection<T> {
+    /**
+     * Retrieves the attribution key associated with the provided offset.
+     */
+    getAtOffset(offset: number): T;
+
+    /**
+     * Retrieve all key/offset pairs stored on this segment. Entries should be ordered by offset, such that
+     * the `i`th result's attribution key applies to offsets in the open range between the `i`th offset and the
+     * `i+1`th offset.
+     * The last entry's key applies to the open interval from the last entry's offset to this collection's length.
+     */
+    getAll(): Iterable<{ offset: number; key: T; }>;
```

Comment:
If we do end up allowing querying for keys of a given kind of attribution, it won't extend to iterable well. But we probably need both anyway.

This API isn't particularly user-friendly right now, it's primarily here so that we can serialize the interface. We could mark it internal with the plans of improving incrementally.

Reply:
I marked it @internal for now. I think we're probably going to want a different public-facing iteration format, but we can add that later. LMK if you feel strongly about further action; resolving this.